include/manager.hpp

```diff
@@ -39,6 +40,117 @@ class Manager
+     * @param[in] i_data - Data to be updated.
+     * @param[in] i_target - Target location to update (0/1/2).
+     *
+     * @return On failure this API throws an error, on success it returns
+     * nothing.
+     */
+    void updateKeyword(const types::Path i_path, const types::VpdData i_data,
+                       const uint8_t i_target);
+
+    /**
+     * @brief Read keyword value.
+     *
+     * API can be used to read VPD keyword.
+     *
+     * To read a keyword from cache, input path should be d-bus object path.
+     * Eg: ("/xyz/openbmc_project/inventory/system/chassis/motherboard").
+     *
+     * To read a keyword from hardware, input path should be EEPROM path.
+     * Eg: ("/sys/bus/i2c/drivers/at24/9-0052/eeprom").
+     *
+     * To read keyword from IPZ type VPD, input data should be in the form of
+     * (Record, Keyword). Eg: ("VINI", "SN").
+     *
+     * To read keyword from keyword type VPD, just keyword name has to be
+     * supplied as input. Eg: ("SN").
+     *
+     * The target to read can either be cache/hardware, whose values are 0/1
+     * respectively.
+     *
+     * @param[in] i_path - D-bus object path/EEPROM path.
+     * @param[in] i_data - Data to be read.
+     * @param[in] i_target - Target location to read from (0/1).
+     *
+     * @return Read value in array of bytes.
+     */
+    types::BinaryVector readKeyword(const types::Path i_path,
+                                    const types::VpdData i_data,
+                                    const uint8_t i_target);
+
+    /**
+     * @brief Collect single FRU VPD
+     * API can be used to perform VPD collection for the given FRU.
+     *
+     * @param[in] i_dbusObjPath - D-bus object path
+     */
+    void collectSingleFruVpd(
+        const sdbusplus::message::object_path& i_dbusObjPath);
+
+    /**
+     * @brief Delete single FRU VPD
+     * API can be used to perform VPD deletion for the given FRU.
+     *
+     * @param[in] i_dbusObjPath - D-bus object path
+     */
+    void deleteSingleFruVpd(
+        const sdbusplus::message::object_path& i_dbusObjPath);
+
+    /**
+     * @brief Get Location code
+     * API can be used to get location code for the given FRU inventory path.
+     *
+     * @param[in] i_dbusObjPath - D-bus object path
+     *
+     * @return FRU Location code
+     */
+    std::string
+        getLocationCode(const sdbusplus::message::object_path& i_dbusObjPath);
+
+    /**
+     * @brief Get Hardware path
+     * API can be used to get EEPROM path for the given inventory path.
+     *
+     * @param[in] i_dbusObjPath - D-bus object path
+     *
+     * @return Corresponding EEPROM path.
+     */
+    std::string getHwPath(const sdbusplus::message::object_path& i_dbusObjPath);
+
+    /**
+     * @brief  Perform VPD recollection
+     * This api will trigger parser to perform VPD recollection for FRUs that
+     * can be replaced at standby.
+     */
+    void performVPDRecollection();
```

Comment:
This is called internally by the class. Should this be private?

Reply:
Ok, we can keep it for now but we need to check if it has a use case for exposing over Dbus in future.